tagger/plot/basic.py

```diff
@@ -28,31 +41,338 @@ def loss_history(plot_dir, history):
+    plt.plot(tpr, fpr, label=f'{class_pair[0]} vs {class_pair[1]} (AUC = {roc_auc:.2f})',
+             color='blue', linewidth=5)
+    plt.grid(True)
+    plt.ylabel('False Positive Rate')
+    plt.xlabel('True Positive Rate')
+    hep.cms.text("Phase 2 Simulation")
+    hep.cms.lumitext("PU 200 (14 TeV)")
+    plt.legend(loc='lower right')
+    plt.yscale('log')
+    plt.ylim([1e-3, 1.1])
+
+    # Save the plot
+    save_path = os.path.join(save_dir, f"ROC_{class_pair[0]}_vs_{class_pair[1]}")
+    plt.savefig(f"{save_path}.pdf", bbox_inches='tight')
+    plt.savefig(f"{save_path}.png", bbox_inches='tight')
+    plt.close()
+
+def ROC(y_pred, y_test, class_labels, plot_dir):
+
+    # Create a colormap for unique colors
+    colormap = cm.get_cmap('Set1', len(class_labels))  # Use 'tab10' with enough colors
+
+    # Create a plot for ROC curves
+    plt.figure(figsize=(16, 16))
+    for i, class_label in enumerate(class_labels):
+
+        # Get true labels and predicted probabilities for the current class
+        y_true = y_test[:, i]  # Extract the one-hot column for the current class
+        y_score = y_pred[:, i] # Predicted probabilities for the current class
+
+        # Compute FPR, TPR, and AUC
+        fpr, tpr, _ = roc_curve(y_true, y_score)
+        roc_auc = auc(fpr, tpr)
+
+        # Plot the ROC curve for the current class
+        plt.plot(tpr, fpr, label=f'{class_label} (AUC = {roc_auc:.2f})',
+                 color=colormap(i), linewidth=5)
+
+    # Plot formatting
+    plt.grid(True)
+    plt.ylabel('False Positive Rate')
+    plt.xlabel('True Positive Rate')
+    hep.cms.text("Phase 2 Simulation")
+    hep.cms.lumitext("PU 200 (14 TeV)")
+    plt.legend(loc='lower right')
 
-def pt_correction_hist(plot_dir, model, data_test):
+    plt.yscale('log')
+    plt.ylim([1e-3, 1.1])
+
+    # Save the plot
+    save_path = os.path.join(plot_dir, "basic_ROC")
+    plt.savefig(f"{save_path}.pdf", bbox_inches='tight')
+    plt.savefig(f"{save_path}.png", bbox_inches='tight')
+    plt.close()
+
+def pt_correction_hist(pt_ratio, truth_pt_test, reco_pt_test, plot_dir):
     """
     Plot the histograms of truth pt, reconstructed (uncorrected) pt, and corrected pt
     """
 
+    plt.figure(figsize=(16, 16))
+    plt.hist(truth_pt_test, bins = 20, range = (0,300), density=True, histtype = 'step', label = 'Truth', linewidth=5)
+    plt.hist(reco_pt_test, bins = 20, range = (0,300), density=True, histtype = 'step', label = 'Reconstructed', linewidth=5)
+    plt.hist(np.multiply(reco_pt_test,pt_ratio), bins = 20, range = (0,300), density=True, histtype = 'step', label = 'NN Predicted', linewidth=5)
+
+    plt.xlabel(r'$p_T$ [GeV]')
+    plt.ylabel('a.u.')
+    plt.legend()  
+    save_path = os.path.join(plot_dir, "pt_hist")
+    plt.savefig(f"{save_path}.pdf", bbox_inches='tight')
+    plt.savefig(f"{save_path}.png", bbox_inches='tight')
+    plt.close()
+
     return
 
-def rms(plot_dir):
+def plot_input_vars(X_test, input_vars, plot_dir):
```

Comment:
It would be useful to have inputs coloured by target class but this can be added in a future PR

Reply:
Could you elaborate on this? Maybe we could open a new issue?